Marlin/src/module/planner.cpp

```diff
@@ -1458,6 +1458,17 @@ void Planner::check_axes_activity() {
 
 #endif // PLANNER_LEVELING
 
+#if ENABLED(FWRETRACT)
+  /**
+   * rz, e - Cartesian positions in mm
+   */
+  void Planner::apply_retract(float &rz, float &e) {
```

Comment:
Every operation in the planner needs to be reversible for the benefit of converting stepper positions back into a cartesian position. So if there's an `apply_retract` there should also be `unapply_retract`. Since we already have `apply_leveling` and `unapply_leveling` the most consistent way to manage it is to move this code into the `apply_leveling` method (or call it from that method) and rename that method `apply_modifiers`. Then rename `unapply_leveling` to `unapply_modifiers` and add code there which removes the hop value. So far the `(un)apply_leveling` methods don't do anything with E, but if generalized to `(un)apply_modifiers` then it would be ok to have E modification there too.

UBL doesn't use `apply_leveling` or `unapply_leveling` at this time, because it was written in its own sandbox, so